api/provider/keycloak.go

```diff
@@ -0,0 +1,94 @@
+package provider
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/netlify/gotrue/conf"
+	"golang.org/x/oauth2"
+)
+
+// Keycloak
+type keycloakProvider struct {
+	*oauth2.Config
+	Host string
+}
+
+type keycloakUser struct {
+	Name          string `json:"name"`
+	Sub           string `json:"sub"`
+	Email         string `json:"email"`
+	EmailVerified bool   `json:"email_verified"`
+}
+
+// NewKeycloakProvider creates a Keycloak account provider.
+func NewKeycloakProvider(ext conf.OAuthProviderConfiguration, scopes string) (OAuthProvider, error) {
+	if err := ext.Validate(); err != nil {
+		return nil, err
+	}
+
+	oauthScopes := []string{
+		"profile",
+		"email",
+	}
+
+	if scopes != "" {
+		oauthScopes = append(oauthScopes, strings.Split(scopes, ",")...)
+	}
+
+	if ext.URL == "" {
+		return nil, errors.New("Unable to find URL for the Keycloak provider")
+	}
+	host := chooseHost(ext.URL, "")
```

Comment:
Hmm i think this line is unnecessary since we are already checking if `ext.URL` is empty above.

Reply:
Yes and no, in terms of checking if its empty yes. chooseHost also removes any trailing slashes. I decided to use chooseHost since most other providers uses it too.

Reply:
Hmm ok i think if you're using it to remove trailing slashes, it's best to just check for it in the same function rather than to use `chooseHost` which was meant for a different purpose.

It might seem unclear in the future and someone unfamiliar with the code will wonder why we default to an empty string.